server/memes/memeController.js

```diff
@@ -0,0 +1,72 @@
+var http = require('http');
+var querystring = require('querystring');
+
+
+module.exports = {
+
+  getMemes: function(request, response, next){
+    var options = {
+      host: 'version1.api.memegenerator.net',
+      path: '/Instances_Select_ByPopular?languageCode=en',
+      method: 'GET'
+    };
+    return http.get(options, function(res){
+      var body = '';
+      res.on('data', function(chunk){
+        body += chunk;
+      });
+      res.on('end', function(){
+        response.status(200).send(body);
+      });
+    });
+  },
+
+  createMeme: function(request, response, next){
+
+    var postData = querystring.stringify({
+      'username': 'memesformankind',
+      'password': 'memesformankind',
+      'generatorID': request.generator_id || 45,
+      'languageCode': 'en',
+      'imageID': request.imageID || 20,
+      'text0': request.text0 || 'toptext',
+      'text1': request.text1 || 'bottomtext'
+    });
+
+    var options = {
+      hostname: 'version1.api.memegenerator.net',
+      path: '/Instance_Create?',
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'Content-Length': postData.length
+      }
+    };
+
+    var req = http.request(options, function(res) {
+      console.log('STATUS: ' + res.statusCode);
+      console.log('HEADERS: ' + JSON.stringify(res.headers));
+      res.setEncoding('utf8');
+      res.on('data', function (chunk) {
+        var body = '';
```

Comment:
body should be outside the data event handler so it and the end event handler can access it via the closure. also need to add the end event handler and have it handle the response.send()